torchrl/envs/libs/smacv2.py

```diff
@@ -377,9 +377,10 @@ def _step(self, tensordict: TensorDictBase) -> TensorDictBase:
         obs = self.get_obs()
         state = self.get_state()
         info = self.observation_spec["info"].encode(info)
-        for info_key in self.observation_spec["info"].keys():
-            if info_key not in info.keys():
-                info[info_key] = self.observation_spec["info"][info_key].zero()
+        info_keys = info.keys()
+        for info_key, spec in self.observation_spec["info"].items():
+            if info_key not in info_keys:
+                info[info_key] = spec.zero()
```

Comment:
```suggestion
        actual_keys = info.keys()
        for expected_key, spec in self.observation_spec["info"].items():
            if expected_key not in actual_keys:
                info[expected_key] = spec.zero()
```
wdyt?
the `info_key in info_keys` is hard to read